mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -473,9 +495,18 @@ ModuleImport::processAliasScopeMetadata(const llvm::MDNode *node) {
       StringAttr description = nullptr;
       if (!aliasScope.getName().empty())
         description = builder.getStringAttr(aliasScope.getName());
-      auto aliasScopeOp = builder.getAttr<AliasScopeAttr>(
-          DistinctAttr::create(builder.getUnitAttr()),
-          cast<AliasScopeDomainAttr>(it->second), description);
+      AliasScopeAttr aliasScopeOp;
+      if (verifySelfRef(scope))
```

Comment:
I would probably also create the id attribute before and then create the scope attribute as above.